webapp/views/App/views/Designer/TaxonomyDetail/store/actions/useUpdate.js

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios'
+import { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+
+import { debounceAction } from '@webapp/utils/reduxUtils'
+import * as Taxonomy from '@core/survey/taxonomy'
+
+import { useSurveyId, TaxonomiesActions } from '@webapp/store/survey'
+
+import { State } from '../state'
+
+export const useUpdate = ({ setState }) => {
+  const dispatch = useDispatch()
+  const surveyId = useSurveyId()
+
+  return useCallback(({ key, value, state }) => {
+    const taxonomy = State.getTaxonomy(state)
+
+    setState(State.assocTaxonomy(State.assocTaxonomyProp({ key, value })(taxonomy))(state))
```

Comment:
maybe some variable would make the code more readable

Reply:
or a pipe

Reply:
sure I also felt the same :)